### PR TITLE
nicer code, --all feature, travis-ci testing, fixes #8

### LINKED ISSUE #8: findings don't get printed
``` python
   if len(hacks_result):
     print "\n"
     for i in hacks_result:
        print "Title     : "+i['Title']
        print "URL       : "+i['URL']
        print "Starts on : "+i['Starts on']
        if i['Ends on']:
           print "Ends on   : "+i['Ends on']
        print "Location  : "+i['location']
        print "-----------------------------------------------------------------"
   if len(hacks_result)==0: # even if this finds something, it won't get printed
     print "Looking for Hackathons in %s, %s..." % (region, country)
     runLists(region)
     runLists(country)
   if len(hacks_result)==0:
     print "Looking for Hackathons in %s..." % (country)
     runLists(country)
   if len(hacks_result)==0:
     print "We couldn't find hackathons in %s, %s.\nTry refining the search location Or try the command, 'hacks locationName'.\nEg: hacks California" % (city, country)
```

the bug is in everything after my comment. I am assigning myself to that bug, I might not finish work on it today, (I am prototyping the socket idea).

### PR DESCRIPTION
**Pull request by mtib**
I added some documentation strings, and moved all the printing to
printResults(), this way we keep our code DRY.
I added the parameter --all or -a, when it is set, the program will print
all found Hackathons. (good when ">"ing to file).

There might be another optimization option, I am questioning whether:

``` python
runLists(country) # does this make L276-278 redundant?
if len(hacks_result) == 0:
    print("Looking for Hackathons in %s..." % (country))
    runLists(country) # can this possibly append something?
```

the last line here could possibly add something to hacks_result...
it's impact on the program isn't massive (timed it with %timeit),
but as we increase sources, it's timeshare might be considerable.

### CODE REVIEW

**Comment by kupiakos** [comment]
If the arguments are going to be made any more complex, why not just use `argparse`?

**Comment by mtib** [comment]
I can't think of any other usage case, where we would need another arg.
But I guess it would make it more future-proof...

**Comment by mtib** [comment]
## I added some travis-ci tests, to make sure all our versions are at least working/running.
- I know the links are wrong, I have no knowledge about how to merge something meant for travis..
- I also added a .gitignore

File: README.md [hunk 1]
```diff
@@ -2,6 +2,8 @@ Hacks
 ============
 A Linux terminal client to find upcoming Hackathons near you or at a particular location.
 
+[![Build Status](https://travis-ci.org/tibyte/Hacks.svg?branch=master)](https://travis-ci.org/tibyte/Hacks)
```
Comment (mtib) [inline README.md:5]: @waseem18 you'd only need to change this line to match your travis-ci setup.

**Comment by waseem18** [comment]
I really liked this PR :) :+1:

**Pull request merged**